### PR TITLE
Fix IDE warnings, cleanup

### PR DESCRIPTION
**Pull request by mkaminski1988**
This PR fixes code warnings flagged by clion—mostly dead code.

### CODE REVIEW

File: src/jsonpath/interpreter.c [hunk 1]
```diff
@@ -107,25 +103,20 @@ void exec_wildcard(zval* arr_head, zval* arr_cur, struct ast_node* tok, zval* re
...
-  zend_ulong num_key;
-
   eval_ast(arr_head, arr_cur, tok, return_value);
 
-  ZEND_HASH_FOREACH_KEY_VAL(HASH_OF(arr_cur), num_key, key, data) {
-    exec_recursive_descent(arr_head, data, tok, return_value);
-  }
+  zval* data;
+
+  ZEND_HASH_FOREACH_VAL_IND(HASH_OF(arr_cur), data) { exec_recursive_descent(arr_head, data, tok, return_value); }
```
Comment (crocodele) [inline src/jsonpath/interpreter.c:115]: Formatting: on three lines would be more readable.

File: src/jsonpath/parser.c [hunk 2]
```diff
@@ -64,15 +58,15 @@ const char* AST_STR[] = {
...
-    "AST_PAREN_LEFT",  /**/
-    "AST_PAREN_RIGHT", /**/
     "AST_RECURSE",     /**/
     "AST_RGXP",        /**/
     "AST_ROOT",        /**/
     "AST_SELECTOR",    /**/
     "AST_WILD_CARD",   /**/
 };
 
+bool is_unary(enum ast_type type) { return type == AST_NEGATION; }
```
Comment (crocodele) [inline src/jsonpath/parser.c:68]: Formatting: function body on its own line would be more readable.

File: src/jsonpath/parser.c [hunk 3]
```diff
@@ -143,9 +137,7 @@ static bool parse_filter_list(PARSER_PARAMS, struct ast_node* tok) {
       /* [:a] => [0:a] */
       /* [a::] => [a:0:] */
       if (slice_count > zend_hash_num_elements(tok->data.d_list.ht)) {
-        if (slice_count == 1) {
-          ht_append_long(tok->data.d_list.ht, INT_MAX);
-        } else if (slice_count == 2) {
+        if (slice_count >= 1 && slice_count <= 2) {
```
Comment (crocodele) [inline src/jsonpath/parser.c:140]: I guess we could make this `if (slice_count == 1 || slice_count == 2) {`.

**Review by crocodele — CHANGES_REQUESTED** [review]
:heart:

File: src/jsonpath/interpreter.c (hunk 1 above)
Comment (mkaminski1988) [reply src/jsonpath/interpreter.c:115]: This is done automatically by clang format. I'll look into updating the rules.

File: src/jsonpath/parser.c (hunk 2 above)
Comment (mkaminski1988) [reply src/jsonpath/parser.c:68]: Same as above, this is automatically formatted by clang-format. Will look into updating the defaults.

File: src/jsonpath/parser.c (hunk 3 above)
Comment (mkaminski1988) [reply src/jsonpath/parser.c:140]: 👍

**Review by crocodele — APPROVED** [review]

**Pull request merged**